vm/vm.go

```diff
@@ -1208,3 +1224,21 @@ func (vm *VM) Fatal(msg string, fields ...zap.Field) {
 	vm.snowCtx.Log.Fatal(msg, fields...)
 	panic("fatal error")
 }
+
+type apiCoreContext struct {
+	*VM
+}
+
+func (c *apiCoreContext) GetDataDir() string {
+	return c.VM.DataDir
+}
```

Comment:
(not necessary for this PR) we should return the data directory that options can use directly ie `filepath.Join(c.VM.DataDir, optNamespace)` rather than leaving this to each option to determine